src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -388,6 +391,13 @@ public int PartialFlushMinSpans
             }
         }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether a span context should be created on exiting a successful Kafka
+        /// Consumer.Consume() call, and closed on entering Consumer.Consume().
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.KafkaCreateConsumerScopeEnabled"/>
+        public bool KafkaCreateConsumerScopeEnabled { get; set; }
```

Comment:
Are we sure we want to make this a public API?

Reply:
I guess that's a more general question - _should_ we expose these toggles as public APIs for customers who are configuring in code?

Reply:
Generally speaking, if we want the setting to be discoverable (Intellisense) and part of the permanent public API, make them public. But if it's meant to be a temporary/experimental feature flag, we keep them internal and tell users to use the env var instead.

Reply:
So this _isn't_ meant to be experimental. Users may need to set it to `false` if they're not using the Kafka library in the "usual" way, i.e. `Consumer.Consume()` in a `while(true)` loop:

https://github.com/DataDog/dd-trace-dotnet/blob/934d97c9fcbbfd750d8dc06343057a72680aa486/test/test-applications/integrations/Samples.Kafka/Consumer.cs#L57-L80

Not really sure _why_ they would need to, but Java have run into situations where it would be required, so good to have the escape hatch

Reply:
Sounds to me like this will stay, then. Thanks!